main.py

```diff
@@ -50,44 +50,39 @@ def _run(self):
                 continue
 
             try:
-                artist_name = TitleParser.get_artist_name_from_submission_title(submission.title)
+                title = TitleParser.create_parsed_submission_from_post_title(submission.title)
             except LookupError:
                 # Can't find artist from submission name, skipping
                 self.database.save_submission(submission)
                 continue
-            
-            # Get artist for most future use
-            prio_artist = TitleParser.get_prioritized_artist(artist_name)
-
-            # Check Youtube.
-            youtube_service = self.youtube.get_service_result(submission.url)
-            if youtube_service.exceeds_threshold is True:
-                self._perform_exceeds_threshold_mod_action(submission, youtube_service)
-                self.database.save_submission(submission)
-                continue
-            
-            # Check Soundcloud.
-            soundcloud_service = self.soundcloud.get_service_result(submission.url)
-            if soundcloud_service.exceeds_threshold is True:
-                self._perform_exceeds_threshold_mod_action(submission, soundcloud_service)
-                self.database.save_submission(submission)
-                continue
 
-            # Check Last.FM
-            try:
-                last_fm_service = self.last_fm.get_service_result(prio_artist)
-                if last_fm_service.exceeds_threshold is True:
-                    self._perform_exceeds_threshold_mod_action(submission, last_fm_service)
-                    self.database.save_submission(submission)
-                    continue
-            except pylast.WSError:
-                # Go ahead and continue execution.  Don't want to fail completely just because one service failed.
-                pass
+            for service in self.services:
+                try:
+                    if isinstance(service, LastFM) is True:
```

Comment:
Maybe add the submission URL to the ParsedSubmission data structure?  Basically, have ParsedSubmission contain *any* relevant info about the post (permalink, user who submitted, etc) so that we can always just pass around that object.

That drops the need for this if statement and allows us some more flexibility in the future for when certain services might need more/less info.

Reply:
Good one!